src/app/components/SheriffDutyBar/SheriffDutyBar.tsx

```diff
@@ -1,43 +1,110 @@
-import * as React from 'react'
+import * as React from 'react';
+import * as moment from 'moment';
 import {
     Glyphicon,
     Label
-} from 'react-bootstrap'
-import './SheriffDutyBar.css'
-import { IdType } from '../../api';
-
+} from 'react-bootstrap';
+import './SheriffDutyBar.css';
+import {
+    IdType,
+    SheriffDuty,
+    AssignmentDuty,
+    WorkSectionCode
+} from '../../api';
+import { getWorkSectionColour } from '../../api/utils';
 
 export interface SheriffDutyBarProps {
     sheriffId?: IdType;
+    sheriffDuty: SheriffDuty;
+    duty: AssignmentDuty;
+    dutyWorkSection?: WorkSectionCode;
     title?: string;
     isExtra?: boolean;
     showBorder?: boolean;
     onRemove?: () => void;
 }
 
 export default class SheriffDutyBar extends React.PureComponent<SheriffDutyBarProps>{
+    getDutyBarWidth() {
```

Comment:
Nice Maths 🤓 !

I think in the future we should try to move some of this up a level into the layout (i.e. the bar list) level, so that could potentially dictate some of the layout using the native css engine (i.e. flexbox etc).